tripy/tripy/backend/compiler_api.py

```diff
@@ -31,6 +32,136 @@
 from tripy.common.shape_bounds import ShapeBounds
 from tripy.frontend import Tensor, Trace
 from tripy.utils import json as json_utils
+from tripy.backend.mlir.utils import MLIRRuntimeClient
+
+
+@export.public_api(document_under="compiler")
+class Stream:
+    """
+    Represents a CUDA stream that can be used to manage concurrent operations.
+
+    This class is a wrapper around the underlying stream object, allowing management of CUDA streams.
+
+    Args:
+        default_stream (bool): If True, uses the default stream. If False, creates a new stream.
+
+    .. code-block:: python
+        :linenos:
+        :caption: Create new streams
+
+        stream_a = tp.Stream()
+        stream_b = tp.Stream()
+        default_stream = tp.Stream(default_stream=True)
+        assert stream_a != stream_b
+        assert default_stream != stream_b
+        stream_a.cleanup()
+        stream_b.cleanup()
+
+    .. code-block:: python
+        :linenos:
+        :caption: Use streams in execution
+
+        linear = tp.Linear(2, 3)
+        compiler = tp.Compiler(linear)
+
+        compiled_linear = compiler.compile(tp.InputInfo((2, 2), dtype=tp.float32))
+        with tp.Stream():
+            a = tp.ones((2, 2), dtype=tp.float32)
+            out = compiled_linear(a)
+
+        assert cp.array_equal(cp.from_dlpack(out), cp.from_dlpack(linear(a)))
+    """
+
+    _active_stream = None
+    _default_stream = None
+    _all_streams = set()
```

Comment:
Why do we need to keep track of all streams? Wouldn't it greatly simplify this class if we didn't do so?

Reply:
It was for syncing all streams but I have removed it from current iteration (we can reintroduce later if required).